node/node.go

```diff
@@ -696,34 +762,29 @@ func (n *Node) makeNodeInfo(nodeID p2p.ID) p2p.NodeInfo {
 		},
 	}
 
-	if n.config.P2P.PexReactor {
+	if config.P2P.PexReactor {
 		nodeInfo.Channels = append(nodeInfo.Channels, pex.PexChannel)
 	}
 
-	rpcListenAddr := n.config.RPC.ListenAddress
-	nodeInfo.Other = append(nodeInfo.Other, fmt.Sprintf("rpc_addr=%v", rpcListenAddr))
+	rpcListenAddr := config.RPC.ListenAddress
+	nodeInfo.Other = append(
+		nodeInfo.Other,
+		fmt.Sprintf("rpc_addr=%v", rpcListenAddr),
+	)
+
+	lAddr := config.P2P.ExternalAddress
 
-	if !n.sw.IsListening() {
-		return nodeInfo
+	if lAddr == "" {
+		lAddr = config.P2P.ListenAddress
 	}
 
-	p2pListener := n.sw.Listeners()[0]
-	p2pHost := p2pListener.ExternalAddressHost()
```

Comment:
Does the new version work with IPv6?

Reply:
Could you be more specific which ipv6 compatibility you refer to?

Reply:
https://github.com/tendermint/tendermint/pull/1818

Reply:
Related to slack chat, we've just transitioned to a IPv6-only SNA. Please ensure IPv6 is supported moving forward. 

TIA :-)

Reply:
@mdyring Rest assured we follow up to harden those concerns, Thanks for staying on top of it.